cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ClientSparkContext.java

```diff
@@ -67,11 +68,22 @@ public ClientSparkContext(Program program, RunId runId, long logicalStartTime, M
 
   @Override
   public <T> T readFromDataset(String datasetName, Class<?> kClass, Class<?> vClass) {
+    return readFromDataset(datasetName, kClass, vClass, Collections.<String, String>emptyMap());
+  }
+
+  @Override
+  public <T> T readFromDataset(String datasetName, Class<?> kClass, Class<?> vClass, Map<String, String> datasetArgs) {
     throw new UnsupportedOperationException("Only supported in SparkProgram.run() execution context");
   }
 
   @Override
   public <T> void writeToDataset(T rdd, String datasetName, Class<?> kClass, Class<?> vClass) {
+    writeToDataset(rdd, datasetName, kClass, vClass, Collections.<String, String>emptyMap());
```

Comment:
Move this implementation to `AbstractSparkContext` so that you don't need to do the same delegation in here and in `ExecutionSparkContext`.